netadapter/server/grpcserver/grpc_connection.go

```diff
@@ -0,0 +1,105 @@
+package grpcserver
+
+import (
+	"net"
+	"sync"
+	"sync/atomic"
+
+	"github.com/pkg/errors"
+
+	"github.com/kaspanet/kaspad/netadapter/server"
+	"github.com/kaspanet/kaspad/netadapter/server/grpcserver/protowire"
+	"github.com/kaspanet/kaspad/wire"
+	"google.golang.org/grpc"
+)
+
+type gRPCConnection struct {
+	server                *gRPCServer
+	address               net.Addr
+	channelWriteLock      sync.Mutex // channelWriteLock makes sure the channels aren't written to after close
```

Comment:
This lock doesn't protect against writes to the channel. It protects against writes to the channel specifically and exclusively when we're shutting down. Please rename it accordingly.